The align_it and align_it_aa methods now belong to the module "Gotoh".

rhliang opened the pull request.

Commit:
The align_it and align_it_aa methods now belong to the module "Gotoh".

Commit:
Added some setup for building a gem containing the Ruby Gotoh bindings.

va7eex commented:
I think the content of this pull request is good, but there are some outstanding questions I have that we can discuss in person on Monday.

Commit:
WIP: copying in the cfe_scripts version of _alignment.rb as a starting point.

This currently doesn't do anything and is here in the interest of
maintaining a sensible "history" for this file.

Commit:
WIP: getting files moved into their new places.

Commit:
Some light refactoring before attempting surgery.

Commit:
WIP: refactoring in progress.

Commit:
WIP: more refactoring in progress.

Some questions have arisen re: the "merge indels to put them into
codons" part; this will require some discussion.

Commit:
The gem now builds and can be 'require'd.

Commit:
Added a stub test file for filling out.

Commit:
Starting to write some of the tests.

Commit:
WIP: continuing to add tests.

Commit:
WIP: more tests written.

Commit:
WIP: fixing tests.  The frame_align tests still don't work.

Commit:
First pass of testing is working.

Commit:
Changed the require statement to hopefully work for both the gem and the tests.

Commit:
WIP: first pass at updating the rakefile.  No idea if it works yet.

Commit:
Cleaned up the rakefile so that the extension is built before tests run.

Commit:
WIP: copied the .gitlab-ci.yml file from the hivdb_algorithm repo.

Commit:
WIP: added the CI configuration from the Phylowatch repo as well.

Commit:
WIP: added some basic devcontainer config and a simple Dockerfile based on cfe_ubuntu.

Commit:
First pass at CI configuration.

Commit:
Tweaking CI configuration.

Commit:
Added more tests for remove_inserts.

These tests were not run on my workstation; they're also an experiment
to see whether CI is working.

Commit:
Fixed a broken test.

This time the tests *were* run on my workstation; the previous commit
showed that the CI pipeline was working correctly.

va7eex commented on ruby/test/templates/index.html.erb:
```diff
@@ -0,0 +1,89 @@
+            <%= title %></h1>
+        <h3>
+        <small>CI Pipeline ID <%= ENV["CI_PIPELINE_ID"] %> from branch <%= ENV["CI_COMMIT_BRANCH"] %>@<%= ENV["CI_PROJECT_NAMESPACE"] %>/<%= ENV["CI_PROJECT_NAME"] %></small></br>
+        <small>Git Commit Hash <%= ENV["CI_COMMIT_SHA"] %> (<%= ENV["CI_COMMIT_SHORT_SHA"] %>)</small>
```
Please check these variables are valid for Github, these look like Gitlab variable names

va7eex commented on ruby/test/test_helper.rb:
```diff
@@ -0,0 +1,30 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+
+if ENV['CI_COMMIT_TAG']
+  run_id = ENV['CI_COMMIT_TAG']
+elsif ENV["CI"]
+  run_id = "#{ENV['CI_PIPELINE_ID']}-#{ENV['CI_COMMIT_SHORT_SHA']}"
```
Please check these variables are valid for Github, these look like Gitlab variable names

va7eex reviewed (commented):
Spotted some variable names we use in our internal gitlab server, please check these are valid github variable names.

Commit:
WIP: some more additions to the CI/CD configuration, and some fixes to the things David noticed in his review.

Commit:
Addressed some of David's findings, and added a CD workflow.

Commit:
Forgot to re-add the files that were renamed.

Commit:
Updated the actions to use newer versions after a warning on Github about using a deprecated Node.js.

Commit:
Fixed a badly-named test, and added a test to check another edge case.

Commit:
Added a step to add the gem as an asset to the release; also streamlined the environment variables.

Commit:
Some updates to the README.
